src/Dialect/ONNX/Transforms/Decompose.cpp

```diff
@@ -791,6 +869,420 @@ bool ShouldDecomposeConvTransposeOpToPhasedConvs(Value convTransposeResult,
+// the alpha value is passed to this method.
+Value decomposeConvT1dIntoPhasedConvs(PatternRewriter &rewriter, Location loc,
+    ONNXConvTransposeOp op, Value convTransposeResult, Value input,
+    Value weights, Value bias, ArrayAttr dilations, IntegerAttr group,
+    ArrayAttr kernel_shape, ArrayAttr pads, ArrayAttr strides,
+    FloatAttr alpha) {
+
+  RankedTensorType weightsType =
+      mlir::cast<RankedTensorType>(weights.getType());
+  assert(weightsType.hasRank() && "Weight tensor must have rank.");
+  Type elementType = getElementType(op.getType());
+  RankedTensorType outputType =
+      mlir::cast<RankedTensorType>(convTransposeResult.getType());
+  auto convTransposeOutputShape = outputType.getShape();
+  auto kernelShape = getIntVectorFromArrayAttr(kernel_shape);
+  auto padsShape =
+      (pads) ? getIntVectorFromArrayAttr(pads) : SmallVector<int64_t>({0, 0});
+  auto stridesShape = (strides) ? getIntVectorFromArrayAttr(strides)
+                                : SmallVector<int64_t>({1});
+
+  int numPhases = stridesShape[0];
+  auto getActivationAppliedToConv = [&](Value conv, Type convOutputType) {
+    if (!alpha)
+      return conv;
+    return (alpha.getValueAsDouble() == 0)
+               ? rewriter.create<ONNXReluOp>(loc, convOutputType, conv)
+                     .getResult()
+               : rewriter
+                     .create<ONNXLeakyReluOp>(loc, convOutputType, conv, alpha)
+                     .getResult();
+  };
+
+  onnx_mlir::MultiDialectBuilder<onnx_mlir::OnnxBuilder> create(rewriter, loc);
+
+  SmallVector<mlir::Value> weightSlices;
+  int step = stridesShape[0];
+  int convKernelSize = kernelShape[0] / stridesShape[0];
+
+  auto axisOnnxConstant = getONNXConstOpFromVector(rewriter, loc, {2});
+  auto stepOnnxConstant = getONNXConstOpFromVector(rewriter, loc, {step});
+  auto weightsShape = weightsType.getShape();
+  auto convWeightsShapedType =
+      weightsType.get({weightsShape[0], weightsShape[1], convKernelSize},
+          weightsType.getElementType());
+  int64_t maxIndex = stridesShape[0];
+  for (int row = 0; row < maxIndex; row++) {
+    int rowEnd = (convKernelSize * step) + row;
+    llvm::SmallVector<int64_t> startVector({row});
+    llvm::SmallVector<int64_t> endVector({rowEnd});
+    auto startOnnxConstant =
+        getONNXConstOpFromVector(rewriter, loc, startVector);
+    auto endOnnxConstant = getONNXConstOpFromVector(rewriter, loc, endVector);
+    weightSlices.push_back(
+        create.onnx.slice(convWeightsShapedType, weights, startOnnxConstant,
+            endOnnxConstant, axisOnnxConstant, stepOnnxConstant));
+  }
+
+  auto convKernelShapeArrayAttr = rewriter.getI64ArrayAttr({convKernelSize});
+
+  // This is the shape of the output from each conv, which contributes to the
+  // final ofm.
+  SmallVector<int64_t> convOutputShape(convTransposeOutputShape);
+  convOutputShape[convOutputShape.size() - 1] =
```

Comment:
NIT: int64_t innermostDim = convOutputShape.size() - 1;
```suggestion
  convOutputShape[convOutputShape.size() - 1] =
```

Reply:
Also isn't `innermostDim` always `2`?

Reply:
True.

Reply:
Next time i will take care of it.